spring-ai-core/src/main/java/org/springframework/ai/vectorstore/observation/VectorStoreObservationContext.java

```diff
@@ -0,0 +1,266 @@
+/*
+* Copyright 2024 - 2024 the original author or authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* https://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+package org.springframework.ai.vectorstore.observation;
+
+import java.util.List;
+
+import org.springframework.ai.document.Document;
+import org.springframework.ai.vectorstore.SearchRequest;
+import org.springframework.util.Assert;
+
+import io.micrometer.observation.Observation;
+
+/**
+ * @author Christian Tzolov
+ * @since 1.0.0
+ */
+
+public class VectorStoreObservationContext extends Observation.Context {
+
+	public enum Operation {
+
+		/**
+		 * VectorStore delete operation.
+		 */
+		ADD("add"),
+		/**
+		 * VectorStore add operation.
+		 */
+		DELETE("delete"),
+		/**
+		 * VectorStore similarity search operation.
+		 */
+		QUERY("query");
+
+		public final String value;
+
+		Operation(String value) {
+			this.value = value;
+		}
+
+	}
+
+	// DELETE
+	private List<String> deleteRequest;
```

Comment:
This list of fields might grow a lot once Spring AI add support for more operations (like for updating embeddings for documents that changed). Consider if they can be grouped into some Records perhaps, or even maybe have a base `VectorStoreObservationContext` (that might be enough for add/delete operations if we remove the need to track the Documents) and specialised implementation whenever relevant (e.g. `QueryVectorStoreObservationContext` for semantic search)

Reply:
Actually intended this initially but then seemed like unnecessary complexity because of the shared context attributes. 
I will rethink it again.